features/update/basic/feature.ts

```diff
@@ -0,0 +1,53 @@
+import { WorkflowUpdateFailedError } from '@temporalio/client';
+import { Feature } from '@temporalio/harness';
+import * as wf from '@temporalio/workflow';
+import * as assert from 'assert';
+
+const WORKFLOW_INITIAL_STATE = '';
+const UPDATE_ARG = 'update-arg';
+const BAD_UPDATE_ARG = 'reject-me';
+const UPDATE_RESULT = 'update-result';
+const UPDATE_RESULT_NOT_YET_RECEIVED = 'update-result-not-yet-received';
+
+const myUpdate = wf.defineUpdate<string, [string]>('myUpdate');
+
+/**
+ * A workflow with a signal and signal validator. If accepted, the signal makes
+ * a change to workflow state. The workflow does not terminate until such a
+ * change occurs.
+ */
+export async function workflow(): Promise<string> {
+  let state = WORKFLOW_INITIAL_STATE;
+  const update = (arg: string) => {
+    state = arg;
+    return UPDATE_RESULT;
+  };
+  const validate = (arg: string) => {
+    if (arg == BAD_UPDATE_ARG) {
+      throw new Error('Invalid Update argument');
+    }
+  };
+  wf.setHandler(myUpdate, update, validate);
```

Comment:
```suggestion
  wf.setHandler(myUpdate, update, { validate });
```

I wonder if it's a bit clearer for anything after the handler to be an options interface instead of just taking validate as the third param. I don't mind this though, just bring up to get others' opinions.

Reply:
Good suggestion. I agree with you and have changed it to use an optional object as last argument instead. I think that's more idiomatic, and much better if we ever need to add additional optional arguments to this function.